pkg/cluster/config/encoding/scheme_test.go

```diff
@@ -17,66 +17,213 @@ limitations under the License.
 package encoding
 
 import (
+	"io/ioutil"
 	"reflect"
 	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime/schema"
 )
 
-// TODO(fabrizio pandini): once we have multiple config API versions we
-// will need more tests
+func TestSplitYAMLDocuments(t *testing.T) {
```

Comment:
and if we have a unified location for the split utility we can avoid some of the tests too.